app/assets/javascripts/discourse/widgets/header.js.es6

```diff
@@ -224,6 +226,11 @@ export default createWidget('header', {
     return h('div.wrap', h('div.contents.clearfix', contents));
   },
 
+  // override to allow plugins to append additional panels
+  additionalPanels() {
```

Comment:
This design pattern will only allow one plugin to override the panels at anyone time. The way that I'll do it is to expose a new API to add panels in [plugin-api](https://github.com/discourse/discourse/blob/747f4812e4323fa303bc2bfc84fcc61f715d7e4a/app/assets/javascripts/discourse/lib/plugin-api.js.es6).